app/models/spree/blog.rb

```diff
@@ -0,0 +1,29 @@
+class Spree::Blog < ActiveRecord::Base
+  include FriendlyId
+  friendly_id :slug_candidates, use: :history
+
+  has_many :blog_entries, dependent: :destroy
+
+  acts_as_paranoid
+
+  before_create :set_slug
+  after_create :set_css_path
+
+  validates_presence_of :name
+  # can add a validation to allow enabling only if AT LEAST ONE blog entry is present
+
+  SLUG_LENGTH = 20
```

Comment:
Did we check if the current set of blog posts across multiple blogs don't have slugs greater than 20 characters in length?